src/highlevel_api.jl

```diff
@@ -164,47 +164,74 @@ num_numa_nodes() = count(hwloc_isa(:NUMANode), gettopology())
 """
 Returns a vector containing the sizes of all available L3 caches in Bytes.
 """
-l3cache_sizes() = map(obj->obj.attr.size, Iterators.filter(obj->obj.type_ == :L3Cache, gettopology()))
+l3cache_sizes() = map(obj -> obj.attr.size, Iterators.filter(obj -> obj.type_ == :L3Cache, gettopology()))
 """
 Returns a vector containing the L3 cache line sizes in Bytes.
 """
-l3cache_linesizes() = map(obj->obj.attr.linesize, Iterators.filter(obj->obj.type_ == :L3Cache, gettopology()))
+l3cache_linesizes() = map(obj -> obj.attr.linesize, Iterators.filter(obj -> obj.type_ == :L3Cache, gettopology()))
 
 """
 Returns a vector containing the sizes of all available L2 caches in Bytes.
 """
-l2cache_sizes() = map(obj->obj.attr.size, Iterators.filter(obj->obj.type_ == :L2Cache, gettopology()))
+l2cache_sizes() = map(obj -> obj.attr.size, Iterators.filter(obj -> obj.type_ == :L2Cache, gettopology()))
 """
 Returns a vector containing the L2 cache line sizes in Bytes.
 """
-l2cache_linesizes() = map(obj->obj.attr.linesize, Iterators.filter(obj->obj.type_ == :L2Cache, gettopology()))
+l2cache_linesizes() = map(obj -> obj.attr.linesize, Iterators.filter(obj -> obj.type_ == :L2Cache, gettopology()))
 
 """
 Returns a vector containing the sizes of all available L1 caches in Bytes.
 """
-l1cache_sizes() = map(obj->obj.attr.size, Iterators.filter(obj->obj.type_ == :L1Cache, gettopology()))
+l1cache_sizes() = map(obj -> obj.attr.size, Iterators.filter(obj -> obj.type_ == :L1Cache, gettopology()))
 """
 Returns a vector containing the L1 cache line sizes in Bytes.
 """
-l1cache_linesizes() = map(obj->obj.attr.linesize, Iterators.filter(obj->obj.type_ == :L1Cache, gettopology()))
+l1cache_linesizes() = map(obj -> obj.attr.linesize, Iterators.filter(obj -> obj.type_ == :L1Cache, gettopology()))
 
 """
+    cachesize()
 Returns the L1, L2, and L3 cache sizes in Bytes.
 (Produces a warning if caches of the same level have different sizes across the system.)
 """
 function cachesize()
     allequal = xs -> all(x == first(xs) for x in xs)
+
     l1 = l1cache_sizes()
     l2 = l2cache_sizes()
     l3 = l3cache_sizes()
 
+    isempty(l1) && throw(ErrorException("Your system doesn't seem to have an L1 cache."))
+    isempty(l2) && throw(ErrorException("Your system doesn't seem to have an L2 cache."))
+    isempty(l3) && throw(ErrorException("Your system doesn't seem to have an L3 cache."))
+
     allequal(l1) || (@warn "Not all L1 cache sizes are equal. Consider using `l1cache_sizes()` instead.")
     allequal(l2) || (@warn "Not all L2 cache sizes are equal. Consider using `l2cache_sizes()` instead.")
     allequal(l3) || (@warn "Not all L3 cache sizes are equal. Consider using `l3cache_sizes()` instead.")
 
     return (L1=first(l1), L2=first(l2), L3=first(l3))
```

Comment:
Suggestion: To provide a defined behavior, I would rather return the largest caches size for each level. Because, on big.LITTLE type architectures, Julia tasks/threads will most likely run on the big cores. Picking first, leaves it to chance, which size is returned.

Reply:
It's not really random (given that hwloc is deterministic) but I'm fine with defaulting to the largest. Since we haven't documented what happens in such a case it's a non-breaking change.